opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -51,48 +55,94 @@ class Meter(APIMeter):
     def __init__(
         self,
         instrumentation_info: InstrumentationInfo,
-        meter_provider: APIMeterProvider,
+        measurement_consumer: MeasurementConsumer,
     ):
         super().__init__(instrumentation_info)
         self._instrumentation_info = instrumentation_info
-        self._meter_provider = meter_provider
+        self._measurement_consumer = measurement_consumer
 
     def create_counter(self, name, unit=None, description=None) -> APICounter:
-        return Counter(self._instrumentation_info, name, unit, description)
+        return Counter(
+            self._instrumentation_info,
+            name,
```

Comment:
nit: can you change this order
```suggestion
        return Counter(
            name,
            self._instrumentation_info,
```

Reply:
Fixed